### ISSUE TITLE
MySQL tinyint(1) coerced to Boolean type unexpectedly

### ISSUE THREAD

**Issue by JamesBream** [JetBrains/Exposed#1272]
I'm having some trouble getting Exposed to handle a MySQL tinyint(1) field properly when trying to run a select query. The Exposed table description is set to expect an integer type as per the MySQL type, but during runtime the RowResult object data field for the relevant index is showing a boolean type which is causing an `IllegalStateException` to be thrown:

```
java.lang.IllegalStateException: Unexpected value of type Int: true of kotlin.Boolean
	at org.jetbrains.exposed.sql.IntegerColumnType.valueFromDB(ColumnType.kt:270)
```

I'm unclear whether I'm doing something wrong here or whether this is a bug/limitation in the way Exposed infers types. I know that MySQL uses `bool` type as a simple alias to `tinyint(1)` under the hood so that could be the confusion there? I've tried to dig into how Exposed does the mapping, but it's a little over my head 😄 

Any thoughts?

**Comment by JamesBream** [comment]
Problem solved, will close this.

For anyone else who runs into this, turns out this is an issue with JDBC and not Exposed. By default (though configurable) JDBC will return you a Boolean type for a `TINYINT` if the storage size is set as 1 because this is basically what the `BOOL` type is using under the hood for storage. See: https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-type-conversions.html

TL;DR? If you use `TINYINT(1)` with JDBC it'll give you a bool by default.
